### PR TITLE
Notification

### PR DESCRIPTION
**Pull request by CreedTech**

### CODE REVIEW

File: src/components/Navbar.tsx [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
 const Navbar: React.FC<INavProps> = ({title,subtitle}) => {
   const classes = useStyles();
+  const [socket, setSocket] = useState<any>("");
   const { activeMenu, isLoggedIn, screenSize, setActiveMenu } = useContextProvider()
 
   const toggleSidebar = () => {
     activeMenu ? setActiveMenu(false) : setActiveMenu(true)
   }
 
+  useEffect(() => {
```
Comment (PhilemonBrain) [inline src/components/Navbar.tsx:16]: change this to env variable

File: src/components/Notification.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    const [notifications, setNotifications] = useState<string[]>([]);
+    const [open, setOpen] = useState<any>(false);
+    const profileId = USER.map(user => {
+        { user.profileId }
+    });
+
+    socket = io('http://localhost:3000');
+    useEffect(() => {
+        socket.on(`newSubscription_${profileId}`, (data: any) => {
+            console.log(data);
```
Comment (PhilemonBrain) [inline src/components/Notification.tsx]: remove all d console.logs

**Review by PhilemonBrain — DISMISSED** [review]

**Comment by deviator53** [comment]
resolve conflicts

File: src/components/Notification.tsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+interface INotificationProps {
+    socket: any;
+};
+
+
+const Notification: React.FC<INotificationProps> = ({ socket }) => {
+    const classes = useStyles();
+    const [notifications, setNotifications] = useState<string[]>([]);
+    const [open, setOpen] = useState<any>(false);
```
Comment (pablo-clueless) [inline src/components/Notification.tsx]: Always set a true or false variable to `<boolean>`, not `<any>`.

File: src/components/Notification.tsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        socket.on(`apiHosted_${user.profileId}`, (data: any) => {
+            setNotifications([...notifications, data])
+        })
+        socket.on(`apiDown_${user.profileId}`, (data: any) => {
+            setNotifications([...notifications, data])
+        })
+    }, [socket]);
+
+
+    const displayNotification = ({ type }: any) => {
```
Comment (pablo-clueless) [inline src/components/Notification.tsx]: Set your type here to be `<number>` since you know it's going to be a number.

**Review by pablo-clueless — COMMENTED** [review]
Please type your variables properly. Unless you're expecting a `Promise`, use a union of the variable and null or undefined, like this:

const [type, setType] = useState<number | null>(0)

**Comment by CreedTech** [comment]
Ok
I'll do that thanks @pablo-clueless

File: src/components/Navbar.tsx [hunk 5]
```diff
@@ -1,15 +1,21 @@
...
+import Notification from './Notification';
+import { io } from 'socket.io-client';
 interface INavProps {
   title?: string
   subtitle?: string
 };
 
 const Navbar: React.FC<INavProps> = () => {
   const classes = useStyles();
+  const [socket, setSocket] = useState<any>("");
```
Comment (taiwoAkindele) [inline src/components/Navbar.tsx:14]: Seems you already know this will a string, so you can set the type to string

**Review by taiwoAkindele — APPROVED** [review]

File: src/components/Navbar.tsx (hunk 5 above)
Comment (CreedTech) [reply src/components/Navbar.tsx:14]: It's not actually collecting a string through
There's a kind of data type associated with sockets that's actually unique and I didn't want to cause errors but I'll still be looking into it
Thanks though

**Review by PhilemonBrain — APPROVED** [review]

**Pull request merged**